src/Tools/dotnet-dev-certs/src/Program.cs

```diff
@@ -115,6 +115,7 @@ public static int Main(string[] args)
                     c.OnExecute(() =>
                     {
                         var reporter = new ConsoleReporter(PhysicalConsole.Singleton, verbose.HasValue(), quiet.HasValue());
+                        bool isInteractive = !Console.IsInputRedirected;
```

Comment:
Not sure how this is related to adding support. Trust is not something that can be run in non-interactive mode, so I think we should avoid it. We haven't checked for the input being redirected in the past and this has not been an issue.

Mac OS will ask for the password to modify the keychain, if that's not possible, then we should do one or two things:
1. Check if we are running under sudo and fail if not when we think we might need admin rights.
2. Check if we are running under sudo and "invoke ourselves" recursively with `sudo`.